lib/tests/mock_server.cc

```diff
@@ -0,0 +1,44 @@
+#include <cstdlib>
+#include <iostream>
+#include <utility>
+#include "mock_server.hpp"
+
+namespace facter {
+
+static void session(tcp::socket sock) {
+  try {
+      for (;;) {
+          char data[1024];
+
+          boost::system::error_code error;
+          sock.read_some(boost::asio::buffer(data), error);
+          if (error == boost::asio::error::eof) {
+              break; // Connection closed cleanly by peer.
+          } else if (error) {
+              throw boost::system::system_error(error); // Some other error.
+          }
+
+          std::string response = "HTTP/1.1 301 Moved Permanently\nContent-length: 0\nLocation: https://puppet.com/\nConnection: close\n\n";
```

Comment:
In my testing it simply returns the content, it didn't follow the redirect. I could change the content of this, just went with something easy to produce.